interfaces/builtin/udisks2.go

```diff
@@ -375,51 +378,69 @@ func (iface *udisks2Interface) Name() string {
 func (iface *udisks2Interface) StaticInfo() interfaces.StaticInfo {
 	return interfaces.StaticInfo{
 		Summary:              udisks2Summary,
+		ImplicitOnClassic:    true,
 		BaseDeclarationSlots: udisks2BaseDeclarationSlots,
 	}
 }
 
 func (iface *udisks2Interface) DBusConnectedPlug(spec *dbus.Specification, plug *interfaces.ConnectedPlug, slot *interfaces.ConnectedSlot) error {
-	spec.AddSnippet(udisks2ConnectedPlugDBus)
+	if !release.OnClassic {
+		spec.AddSnippet(udisks2ConnectedPlugDBus)
+	}
 	return nil
 }
 
 func (iface *udisks2Interface) DBusPermanentSlot(spec *dbus.Specification, slot *snap.SlotInfo) error {
-	spec.AddSnippet(udisks2PermanentSlotDBus)
+	if !release.OnClassic {
+		spec.AddSnippet(udisks2PermanentSlotDBus)
+	}
 	return nil
 }
 
 func (iface *udisks2Interface) AppArmorConnectedPlug(spec *apparmor.Specification, plug *interfaces.ConnectedPlug, slot *interfaces.ConnectedSlot) error {
 	old := "###SLOT_SECURITY_TAGS###"
-	new := slotAppLabelExpr(slot)
+	var new string
+	if release.OnClassic {
+		new = "unconfined"
+	} else {
+		new = slotAppLabelExpr(slot)
+	}
 	snippet := strings.Replace(udisks2ConnectedPlugAppArmor, old, new, -1)
 	spec.AddSnippet(snippet)
 	return nil
 }
 
 func (iface *udisks2Interface) AppArmorPermanentSlot(spec *apparmor.Specification, slot *snap.SlotInfo) error {
-	spec.AddSnippet(udisks2PermanentSlotAppArmor)
+	if !release.OnClassic {
```

Comment:
This is fine because it applies to the slot (service) and udisks is not confined on classic systems.

Interestingly if a distribution ever confines udisks itself we have no way to deal with that easily.

Reply:
Yes, see my previous point. This is known. On a distro level, we special case this based on distro. We could potential call libapparmor to try to dynamically detect the confinement of the service on classic. It needs more thought.